### PR TITLE
IGNITE-16182: moved TcpDiscoveryZookeeperIpFinder to extensions.

### PR DESCRIPTION
**Pull request by Sega76**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/zookeeper/pom.xml [hunk 1]
```diff
@@ -41,48 +41,11 @@
...
-            <version>${curator.version}</version>
-        </dependency>
-
-        <dependency>
-            <groupId>org.apache.curator</groupId>
-            <artifactId>curator-recipes</artifactId>
-            <version>${curator.version}</version>
-        </dependency>
-
         <dependency>
```
Comment (ivandasch) [inline modules/zookeeper/pom.xml:44]: We don't need it in release

We need only `curator-test` in testing dependency

Comment (Sega76) [reply modules/zookeeper/pom.xml:44]: @ivandasch this used with scope tests and will not in release

File: modules/zookeeper/pom.xml [hunk 2]
```diff
@@ -45,44 +45,7 @@
             <groupId>org.apache.curator</groupId>
```
Comment (ivandasch) [inline modules/zookeeper/pom.xml:45]: We don't need this at all
All we need is `curator-test`

It is defined few lines after.

Comment (Sega76) [reply modules/zookeeper/pom.xml:45]: `curator-framework` need for ZookeeperDiscoverySpiTestBase

**Review by NSAmelchev — APPROVED** [review]

**Pull request merged**